### PR TITLE
doc: Introduce Fast Pair documentation

### PR DESCRIPTION
**Pull request by MarekPieta**
Change introduces documentation for Fast Pair in the NCS. Documentation is also added for sample and provisioning script.

Jira: NCSDK-14165

### CODE REVIEW

File: doc/nrf/libraries/bluetooth_services/services/fast_pair.rst [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+**********************
+
+The implementation uses :c:macro:`BT_GATT_SERVICE_DEFINE` to statically define and register the Fast Pair GATT service.
+The service automatically handles all of the requests received from the Fast Pair Seeker.
+No application input is required to handle the requests.
+
+Bluetooth authentication
+========================
+
+The service overrides the set of Zephyr's Bluetooth authentication callbacks (:c:struct:`bt_conn_auth_cb`) while handling Key-Based Pairing request.
```
Comment (miha-nordic) [inline doc/nrf/libraries/bluetooth_services/services/fast_pair.rst:98]: Don't this callbacks change in latest upmerge?

Comment (miha-nordic) [inline doc/nrf/libraries/bluetooth_services/services/fast_pair.rst:98]: Don't this callbacks change in latest upmerge?

**Review by miha-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/bluetooth_services/services/fast_pair.rst (hunk 1 above)
Comment (MarekPieta) [reply doc/nrf/libraries/bluetooth_services/services/fast_pair.rst:98]: Yes, the authentication info callback (`bt_conn_auth_info_cb`) were separated. Other callbacks that were part of `bt_conn_auth_cb` are still working in the same way.
I aligned to new callbacks in Fast Pair implementation while rebasing

**Review by miha-nordic — APPROVED** [review]

**Review by alstrzebonski — CHANGES_REQUESTED** [review]

**Review by greg-fer — CHANGES_REQUESTED** [review]
Please check my commit and squash it. We need to make sure the images are properly scaled before we merge this PR.

File: samples/bluetooth/peripheral_fast_pair/README.rst [hunk 2]
```diff
@@ -7,13 +7,36 @@ Bluetooth: Fast Pair
    :local:
    :depth: 2
 
-The Fast Pair sample demonstrates how to use the Google Fast Pair Service in the |NCS|.
+The Bluetooth® Fast Pair sample demonstrates :ref:`ug_bt_fast_pair`.
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply samples/bluetooth/peripheral_fast_pair/README.rst]: Ok, changed

File: doc/nrf/libraries/bluetooth_services/services/fast_pair.rst [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+Because of this requirement, the default values of the following Kconfig options are modified by the GFPS Kconfig:
+
+* :kconfig:option:`CONFIG_BT_L2CAP_TX_MTU`
+* :kconfig:option:`CONFIG_BT_BUF_ACL_TX_SIZE`
+* :kconfig:option:`CONFIG_BT_BUF_ACL_RX_SIZE`
+* :kconfig:option:`CONFIG_BT_CTLR_DATA_LENGTH_MAX`
+
+.. tip::
+   In case of the nRF53, this part of the configuration cannot be automatically updated for the network core and it must be manually aligned by the user.
+   These options on the network core shall be set to default values specified by GFPS Kconfg.
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply doc/nrf/libraries/bluetooth_services/services/fast_pair.rst]: Fixed by rephrase introduced by @greg-fer

File: scripts/nrf_provision/fast_pair/README.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+     -h, --help            show this help message and exit
+     -o OUT_FILE, --out_file OUT_FILE
+                           Name of the output file
+     -a ADDRESS, --address ADDRESS
+                           Address of provisioning partition start (in hex)
+     -m MODEL_ID, --model_id MODEL_ID
+                           Model ID (in format 0xXXXXXX)
+     -k ANTI_SPOOFING_KEY, --anti_spoofing_key ANTI_SPOOFING_KEY
+                           Anti Spoofing Key (base64 encoded)
+
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply scripts/nrf_provision/fast_pair/README.rst:69]: @greg-fer can we do something about it?

**Review by greg-fer — APPROVED** [review]

**Comment by greg-fer** [comment]
@MarekPieta , remember to check changelog for relevant entries :)

**Review by greg-fer — APPROVED** [review]

**Review by alstrzebonski — APPROVED** [review]

**Pull request merged**